doc: Add redis example and TypeScript definition (#12)
jsumners/fastify-server-session: Readme.md

+ [fastify-caching](https://www.npmjs.com/package/fastify-caching): used to
store the session data via the `fastify.cache` decorator.

**Installation:**

```
npm install fastify-server-session fastify-cookie fastify-caching

> ```

Reviewer: Do we really need this? I think people understand how to manage packages at this point.

Author: I personally find it's a good pattern as there is other dependencies. Also some repositories have Github repository names that does not equal the npm package name. Avoid potential confusion in a sense

## Example

### Server Side Storage

> Using this implementation the sessions will be stored in the Fastify instance
> making the server statefull.

Reviewer: ```suggestion
Using this implementation the sessions will be stored in memory on the Fastify instance
making the server stateful. This is not recommended for production. It will not share
state among multiple instances of the same application.
```

})
```

### Remote Cache Store

> `fastify-caching` offers the connectivity to a remote store like shown below with `ioredis` and `abstract-cache`.
> See `fastify-caching` [documentation](https://github.com/fastify/fastify-caching)

Reviewer: ```suggestion
`fastify-caching` offers the connectivity to a remote store as shown below with `ioredis` and `abstract-cache`.
See `fastify-caching` [documentation](https://github.com/fastify/fastify-caching) for other
storage capabilities.
```

```js
// This example requires the following packages to be installed
// - ioredis
// - abstract-cache

const IORedis = require('ioredis')
const redis = new IORedis({host: '127.0.0.1'})
const abcache = require('abstract-cache')({

>   useAwait: false,

Reviewer: ```suggestion
  useAwait: true,
```

Sorry. This actually a mistake in the upstream documentation that I haven't gotten around to fixing. The `abstract-cache-redis` module implements the `await` protocol.

  driver: {
    name: 'abstract-cache-redis',
    options: {client: redis}
  }
})
const fastify = require('fastify')()
fastify
  .register(require('fastify-cookie'))
  .register(require('fastify-caching'), {cache: abcache})
  .register(require('fastify-server-session'), {
    secretKey: 'some-secret-password-at-least-32-characters-long',
    sessionMaxAge: 900000, // 15 minutes in milliseconds
    cookie: {
      domain: '.example.com',
      path: '/'
    }
  })

fastify.get('/one', (req, reply) => {
  req.session.foo = 'foo'
  reply.send()
})

fastify.get('/two', (req, reply) => {
  reply.send({foo: req.session.foo})
})
```

> **Note:** In the previous example the `sessionMaxAge` value will set the redis TTL of the session key.

Reviewer: ```suggestion
**Note:** In the previous example the `sessionMaxAge` value will set the Redis TTL of the session key.
```

[cookiedoc]: https://www.npmjs.com/package/cookie#options-1

## TypeScript

> To use the type checking on session object you can use the declaration:

Reviewer: ```suggestion
To use type checking on session object you can use the declaration:
```
